pkg/k8s/deployment.go

```diff
@@ -1,22 +1,25 @@
 package k8s
 
 import (
+	"context"
+
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	k8sappsv1 "k8s.io/client-go/kubernetes/typed/apps/v1"
 )
 
 // DeleteDeployment deletes the deployment given using the client given
-func DeleteDeployment(name string, cl k8sappsv1.DeploymentInterface) error {
-	return cl.Delete(name, &metav1.DeleteOptions{})
+func DeleteDeployment(ctx context.Context, name string, cl k8sappsv1.DeploymentInterface) error {
```

Comment:
@khaosdoctor can you update this to use `client.Client` instead of the `DeploymentInterface`?

Reply:
Added on 877d2